Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -0,0 +1,314 @@
+﻿using Kudu.Contracts.Tracing;
+using Kudu.Core.Infrastructure;
+using Kudu.Core.Tracing;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace Kudu.Services.Performance
+{
+    internal static class ProfileManager
+    {
+        private const int ProcessExitTimeoutInSeconds = 20;
+
+        private static ConcurrentDictionary<int, ProfileInfo> _profilingList = new ConcurrentDictionary<int, ProfileInfo>();
+        private static object _lockObject = new object();
+
+        // The profiling session timeout, this is temp fix before VS2015 Update 1.
+        private static readonly TimeSpan _profilingTimeout = TimeSpan.FromMinutes(15);
+        private static Timer _profilingIdleTimer;
+
+        private static string _processName = System.Environment.ExpandEnvironmentVariables("%SystemDrive%\\msvsmon\\profiler\\VSStandardCollector.Dev14.exe");
+
+        internal static async Task<HttpResponseInfo> StartProfileAsync(int processId, ITracer tracer = null)
```

Comment:
per my other comment we shouldn't throw exceptions to control flow. here is an example:

                // check if the profiling is running for the given process. If it doesn't return 404.
                if (!_profilingList.ContainsKey(processId))
                {
                    return new HttpResponseInfo(HttpStatusCode.NotFound, string.Format("Profiling for process '{0}' is not running.", processId));
                }

why would we throw exception here?